src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -225,6 +229,19 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
             builder.RequestDelegate = factoryCreatedRequestDelegate;
         }
 
+        for (var i = entry.FinalConventions.Count - 1; i >= 0; i--)
+        {
+            entry.FinalConventions[i](builder);
+        }
+
+        if (groupFinalConventions is not null)
+        {
+            for (var i = groupFinalConventions.Count - 1; i >= 0; i--)
+            {
+                groupFinalConventions[i](builder);
+            }
+        }
```

Comment:
It would be convenient if we could run the final conventions after running the normal conventions for *all* the endpoints produced by this data source. That way conventions could be made aware of all the endpoints provided by the EndpointDataSource before finalizing the endpoints if the care. And I don't see a downside to running them this way.

Reply:
I think this is punt-worthy given the ✨ deadline ✨  but I've captured an issue for it in https://github.com/dotnet/aspnetcore/issues/43246.

Reply:
Pro PR move 😆. I cannot come up with a real scenario for this yet, so this seems fair. It was more that if it were easy enough, it would be nice to have.